lib/views/main_screen.dart

```diff
@@ -1,52 +1,277 @@
+          "62ccfccd3eb7fd2a30f41601",
+          "<YOUR_ORG_ID_FOR_DEVELOPMENT>"
+        ]
+      },
+      {
+        "_id": "63929c1915ad470d13c3bcb6",
+        "pluginInstallStatus": true,
+        "pluginName": "Posts",
+        "pluginCreatedBy": "Mr.Johnathan Doe ",
+        "pluginDesc":
+            "Presenting the Donation plugin for the talawa app that enables members of the organization to  do One time or Reccurinng donations to an  organization",
+        "installedOrgs": [
+          "62ccfccd3eb7fd2a30f41601",
+          "62ccfccd3eb7fd2a30f41601",
+          "<YOUR_ORG_ID_FOR_DEVELOPMENT>"
+        ]
+      },
+      {
+        "_id": "62cfcf71824dc26e1fbeed46",
+        "pluginInstallStatus": true,
+        "pluginName": "Donation",
+        "pluginCreatedBy": "Kimia",
+        "pluginDesc":
+            "See all available organizations in entire world using this plugins",
+        "installedOrgs": [
+          "62ccfccd3eb7fd2a30f41601",
+          "<YOUR_ORG_ID_FOR_DEVELOPMENT>"
+        ]
+      },
+      {
+        "_id": "62cfcf71824dc26e1fbeed46",
+        "pluginInstallStatus": true,
+        "pluginName": "Chats",
+        "pluginCreatedBy": "Kimia",
+        "pluginDesc":
+            "See all available organizations in entire world using this plugins",
+        "installedOrgs": [
+          "62ccfccd3eb7fd2a30f41601",
+          "<YOUR_ORG_ID_FOR_DEVELOPMENT>"
+        ]
+      }
+    ];
+
+    /// Here we are dynamically adding BottomNavigationBarItems in navbar by mapping over the data received from the server.
+    // ignore: avoid_dynamic_calls
+    pluginList.forEach((plugin) => {
+          if (navNameClasses.containsKey(plugin["pluginName"] as String) &&
+              plugin["pluginInstallStatus"] as bool)
+            {
+              navBarItems.add(BottomNavigationBarItem(
+                icon: navNameIcons[plugin["pluginName"]] ??
+                    const Icon(Icons.home),
+                label: AppLocalizations.of(context)!
+                    .strictTranslate(plugin["pluginName"] as String),
+              )),
+              navBarClasses.add(navNameClasses[plugin["pluginName"]]["class"]
+                  as StatelessWidget),
+            },
+        });
+
+    /// updating the state to re-render the navbar widget.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      /// This line causes the app to continiously check for plugins if they are updated
+      /// and it will automatically re-render the navbar with enables or disabled features
+      // Timer? timer;
+      // timer = Timer.periodic(
+      //   const Duration(seconds: 15),
+      //   (Timer t) => FetchPluginList(),s
+      // );
+
+      // FetchPluginList(); // dont delete
+      setState(() {});
+    });
+
+    return [navBarItems, navBarClasses];
+  }
+
   @override
   Widget build(BuildContext context) {
+    ///Features that are not meant to be implemented as plugins should be kept here.
+    navBarItems = [
```

Comment:
Having static data in the build method is not a good idea. Please consider refactoring this.

Reply:
That's how we planned to do it, previously

Reply:
We can improve now